pkg/oc/cli/cmd/set/deployment_strategy.go

```diff
@@ -0,0 +1,483 @@
+			continue
+		}
+
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, types.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			fmt.Fprintf(o.Err, "error: %v\n", err)
+			failed = true
+			continue
+		}
+
+		info.Refresh(obj, true)
+		kcmdutil.PrintSuccess(o.Mapper, o.ShortOutput, o.Out, info.Mapping.Resource, info.Name, false, "updated")
+	}
+	if failed {
+		return kcmdutil.ErrExit
+	}
+	return nil
+}
+
+func (o *DeploymentStrategyOptions) applyDeploymentConfigRecreateStrategyOptions(s *deployapi.RecreateDeploymentStrategyParams) bool {
+	if o.TimeoutSeconds != deployapi.DefaultRecreateTimeoutSeconds {
+		s.TimeoutSeconds = &o.TimeoutSeconds
+		return true
+	}
+	return false
+}
+
+func (o *DeploymentStrategyOptions) applyDeploymentConfigRollingStrategyOptions(s *deployapi.RollingDeploymentStrategyParams) bool {
+	updated := false
+	if o.TimeoutSeconds != deployapi.DefaultRecreateTimeoutSeconds {
+		s.TimeoutSeconds = &o.TimeoutSeconds
+		updated = true
+	}
+	if len(o.MaxSurge) > 0 {
+		s.MaxSurge = intstr.FromString(o.MaxSurge)
+		updated = true
+	}
+	if len(o.MaxUnavailable) > 0 {
+		s.MaxUnavailable = intstr.FromString(o.MaxUnavailable)
+		updated = true
+	}
+	// TODO: This has potential to reset the pre-existing values for these two
+	// fields if they are already set to custom values. We need to pick if this
+	// command will override those and set them back to default values if used or
+	// not.
+	if o.IntervalSeconds != deployapi.DefaultRollingIntervalSeconds || (s.IntervalSeconds != nil && *s.IntervalSeconds != deployapi.DefaultRollingIntervalSeconds) {
+		s.IntervalSeconds = &o.IntervalSeconds
+		updated = true
+	}
+	if o.UpdatePeriodSeconds != deployapi.DefaultRollingUpdatePeriodSeconds || (s.UpdatePeriodSeconds != nil && *s.UpdatePeriodSeconds != deployapi.DefaultRollingUpdatePeriodSeconds) {
+		s.UpdatePeriodSeconds = &o.UpdatePeriodSeconds
+		updated = true
+	}
+
+	return updated
+}
+
+func (o *DeploymentStrategyOptions) applyDeploymentConfigCustomStrategyOptions(s *deployapi.CustomDeploymentStrategyParams) bool {
+	updated := false
+	if len(o.Command) > 0 {
+		s.Command = o.Command
+		updated = true
+	}
+	if len(o.Environment) > 0 {
+		env, _, err := cmdutil.ParseEnv(o.Environment, nil)
+		if err == nil {
+			s.Environment = env
+		}
+		updated = true
+	}
+	if len(o.Image) > 0 {
+		s.Image = o.Image
+		updated = true
+	}
+	return updated
+}
+
+func (o *DeploymentStrategyOptions) updateDeploymentConfig(dc *deployapi.DeploymentConfig) (bool, error) {
+	var (
+		strategyChanged bool
+		paramsChanged   bool
+	)
+	if o.Recreate {
+		dc.Spec.Strategy.CustomParams = nil
+		if dc.Spec.Strategy.RecreateParams == nil {
+			dc.Spec.Strategy.RecreateParams = &deployapi.RecreateDeploymentStrategyParams{}
+			strategyChanged = true
+		}
+		if dc.Spec.Strategy.RollingParams != nil {
+			// Preserve previous strategy parameters that are applicable
```

Comment:
this have potential to be useful and confusing at the same time. one might expect that he gets default values for that strategy

Reply:
I commented in other place, we need to decide if we want to preserve fields... I would assume that when you switching from rolling to recreate, you want to keep the hooks. Maybe we should have special flags, like `--copy-hooks` or something?

Reply:
yeah, we need flag to control it. not sure about the default

Reply:
this was solved by .Change() I believe... I only mutate if user request to mutate, otherwise just use the defaults...